### PR TITLE
fix(328): Update to only have a single submit

### PR DESCRIPTION
**Pull request by knguyenrise8**
## Description
 - improve user flow to only have single submit
 
## Screenshots (if applicable)

Oct-24-2024 10-54-24

## Related Issues
[Link any related issues or tasks from your project management system.]

## Checklist
- [ ] The title of this PR is descriptive and concise.
- [ ] My changes follow the style guidelines of this project.
- [ ] I have added or updated test cases to cover my changes.
- [ ] I've let the team know about this PR by linking it in the review channel

### CODE REVIEW

File: frontend/src/pages/ExtractProcess.tsx [hunk 1]
```diff
@@ -112,11 +112,13 @@ const ExtractProcess = () => {
...
 
-    convertPdfToImages(pdfFile).then((imgs) => {
+    convertPdfToImages(pdfFile)
+    .then((imgs) => {
       setImages(imgs);
       localStorage.setItem("images", JSON.stringify(imgs));
-    });
-  }, [files]);
+    })
+    .finally(() => handleSubmit())
```
Comment (schreiaj) [inline frontend/src/pages/ExtractProcess.tsx:120]: Should this be a `finally` or another `then`? 

Do we want to call `handleSubmit` in the event of an error?

File: frontend/src/components/TemplatesIndex/TemplatesIndex.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ export const TemplatesIndex: FC<TemplateIndexProps> = () => {
         'lastUpdated': (d) => {
             const date = Date.parse(d)
             if (isNaN(date)) {
-                return "Invalid Date"
+                return new Date().toLocaleDateString()
```
Comment (schreiaj) [inline frontend/src/components/TemplatesIndex/TemplatesIndex.tsx:46]: I still have heart burn about bundling unrelated changes.

**Review by schreiaj — COMMENTED** [review]
If calling `handleSubmit` even on failure is the intended behavior, it looks fine.

File: frontend/src/pages/ExtractProcess.tsx (hunk 1 above)
Comment (knguyenrise8) [reply frontend/src/pages/ExtractProcess.tsx:120]: I think Pri said we are gonna handle errors later. Right now we are focusing on happy path. I’m looking to using a error handling tool like chakra UI’s toast or modal for retries

**Review by schreiaj — APPROVED** [review]
